### DISCUSSION TITLE
Enable different CORS for different users

### DISCUSSION THREAD

**Discussion by gikpro** [fastapi/fastapi#5505] [Questions] [answered]
I am working on API project, i need to know the best way to enable different CORS for different users. For example: users that have a Free plan will have only access to the API if their requests come from localhost or 127.0.0.1, and users that use Premium plan will have access from any server(computer).

**Comment by iudeen** [comment]
You may modify the middleware as per your needs. You can do by overriding the `__call__` method in https://github.com/encode/starlette/blob/master/starlette/middleware/cors.py#L74

**Reply by gikpro** [reply]
Thanks for your reply, my problem now is where i must begin :-D Can i see an example ?

**Comment by odiseo0** [comment, marked as answer]
Probably you should use another approach to this problem, implementing Authorization in your API for example.

I wouldn't recommend using CORS, because CORS only works in client-side

**Reply by tiangolo** [reply]
Agreed ☝️
